src/trie/intermediate_hashes.rs

```diff
@@ -282,18 +289,269 @@ where
             self.can_skip_state = self.stack.last().unwrap().hash_flag();
         }
     }
+
+    fn changed_mut(&mut self) -> &mut PrefixSet {
+        self.changed
+    }
+}
+
+struct DbTrieLoader<'db, 'tx, 'tmp, 'co, 'nc, Tx>
+where
+    Tx: MutableTransaction<'db>,
+    'db: 'tx,
+    'tmp: 'co,
+    'co: 'nc,
+{
+    txn: &'tx mut Tx,
+    hb: HashBuilder<'nc>,
+    storage_collector: RefCell<&'co mut TableCollector<'tmp, tables::TrieStorage>>,
```

Comment:
Make `calculate_storage_root` accept `&mut self` instead

Reply:
That will be part of the refactoring. The `RefCell` here and one or two other hacks were necessary to follow silkworm closely without having two mutable refs at once.